src/AzSK.ADO/Framework/Core/SVT/SVTResourceResolver.ps1

```diff
@@ -453,7 +451,7 @@ class SVTResourceResolver: AzSKRoot {
                         serviceConnections = -1;
                     };
 
-                    if($this.UsePartialCommits -and $this.DoNotRefetchResources){
+                    if($this.UsePartialCommits -and $this.DoNotRefetchResources -and $this.isDnrrAllProject -eq $false){
```

Comment:
Why $this.isDnrrAllProject -eq $false in if flock if its getting changed in the next block only?

$this.isDnrrAllProject will always be false here

Reply:
Added this check to stop printing "Resuming scan from last commit. Fetching unscanned resources..." for the rest of the projects. Also when the loop starts for next project we do not need to get non scanned resources again. The tracker file is parsed in one go.